File: sample/ruby/main_texture_sample.rb
require 'voxelamming'
# require_relative 'voxelamming'

texture_names = ["grass", "stone", "dirt", "planks", "bricks"]
room_name = "1000"
vox = Voxelamming::VoxelammingManager.new(room_name)
# vox = VoxelammingManager.new(room_name)

vox.set_box_size(1)
vox.set_build_interval(0.01)

texture_names.each_with_index do |texture, i|
  vox.create_box(0, texture_names.length - i - 1, 0, texture: texture)
end

vox.send_data()
vox.clear_data()
sleep(0.1)

vox.set_box_size(1)
vox.set_build_interval(0.01)
vox.change_shape('sphere')
texture_names.each_with_index do |texture, i|
  vox.create_box(1, texture_names.length - i - 1, 0, texture: texture)
end

vox.send_data()
vox.clear_data()
sleep(0.1)

vox.set_box_size(1)
vox.set_build_interval(0.01)
vox.change_shape('plane')
texture_names.each_with_index do |texture, i|
  vox.create_box(2, texture_names.length - i - 1, 0, texture: texture)
end

vox.send_data()
vox.clear_data()

File: sample/python/main_texture_sample.py
from time import sleep
# voxelammingパッケージからVoxelammingクラスをインポートします
from voxelamming import Voxelamming
# from voxelamming_local import Voxelamming  # ローカルで開発している場合はこちらを使う

texture_names = ["grass", "stone", "dirt", "planks", "bricks"]
# Voxelammingアプリに表示されている部屋名を指定してください
room_name = "1000"
# Voxelammingクラスのインスタンスを生成します
voxelamming = Voxelamming(room_name)

# ボクセルの設定を行います
voxelamming.set_box_size(1)
voxelamming.set_build_interval(0.01)

# ボクセルを配置するため、位置と色を設定します
for i, texture in enumerate(texture_names):
    voxelamming.create_box(0, len(texture_names) - i - 1, 0, texture=texture)

# ボクセルデータをアプリに送信します。
voxelamming.send_data()
# ボクセルデータをクリアします。
voxelamming.clear_data()
sleep(0.1)

# ボクセルの設定を行います
voxelamming.set_box_size(1)
voxelamming.set_build_interval(0.01)
voxelamming.change_shape('sphere')

# ボクセルを配置するため、位置と色を設定します
for i, texture in enumerate(texture_names):
    voxelamming.create_box(1, len(texture_names) - i - 1, 0, texture=texture)

# ボクセルデータをアプリに送信します。
voxelamming.send_data()
# ボクセルデータをクリアします。
voxelamming.clear_data()
sleep(0.1)

# ボクセルの設定を行います
voxelamming.set_box_size(1)
voxelamming.set_build_interval(0.01)
voxelamming.change_shape('plane')

# ボクセルを配置するため、位置と色を設定します
for i, texture in enumerate(texture_names):
    voxelamming.create_box(2, len(texture_names) - i - 1, 0, texture=texture)

# ボクセルデータをアプリに送信します。
voxelamming.send_data()
